### PR TITLE
Adding Configure basemap style language sample

### PR DESCRIPTION
**Pull request by har13205**
# Description

This PR is to add `Configure basemap style language` sample

## Type of change

- [ ] Bug fix
- [x] New sample implementation
- [ ] Sample viewer enhancement
- [ ] Other enhancement

**Platforms tested on**:

- [x] Windows
- [x] Android
- [ ] Linux
- [x] macOS
- [ ] iOS

## Checklist

- [ ] Runs and compiles on all active platforms as a standalone sample
- [x] Runs and compiles in the sample viewer(s)
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [ ] Self-review of changes
- [ ] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [x] Code is commented with correct formatting (CTRL+i)
- [ ] All variable and method names are camel case
- [ ] There is no leftover commented code
- [x] Screenshots are correct size and display in description tab (500px by 500px, platform agnostic)
- [x] If adding a new sample, it is added to the sample viewer
- [ ] Cherry-picked to Main branch (if applicable)

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.qml
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.qml:131]:
```suggestion
        model.setNewBasemap(globalButton.checked, comboBox.currentText);
```

Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.qml:131]: Probably both should have a clearer name.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/screenshot.png
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/screenshot.png:1]: The screenshot proportions don't look correct. It looks like it's been squished.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:94]:
```suggestion
  // A SpecificLanguage setting overrides the LanguageStrategy settings
```

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:92]:
```suggestion
  basemapStyleParameters->setLanguageStrategy(global ? BasemapStyleLanguageStrategy::Global : BasemapStyleLanguageStrategy::Local);
```

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:95]: Review bracket style.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.qml
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.qml:130]: seems unnecessary to have the JS function wrapping the C++ function - why not just directly call the C++ function from the combobox itself?

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:105]:
```suggestion
  } 
  else if (language == "Bulgarian")
  {
    basemapStyleParameters->setSpecificLanguage("bg");
  } 
  else if (language == "Greek")
  {
    basemapStyleParameters->setSpecificLanguage("el");
  } 
  else if (language == "Turkish")
  {
```

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:89]: It looks like this is always needed. If so, consider creating it in the constructor.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:84]:
```suggestion
// Set new basemap language based on the parameters selected
```

**Review by JamesMBallard — COMMENTED** [review]
Looking good, but I noticed a few other minor things.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.h
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.h:54]: by convention, member variables should start with `m_`

```suggestion
  Esri::ArcGISRuntime::BasemapStyleParameters* m_basemapStyleParameters = nullptr;
  Esri::ArcGISRuntime::Basemap* m_basemap = nullptr;
```

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:59]: Seems like we should initialize the spatial reference of the map, since we are immediately setting it on a map view and then setting a viewpoint. I'm actually surprised it works to set a viewpoint when there's no spatial reference set.

```suggestion
  m_map = new Map(SpatialReference::webMercator());
```

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:109]: since this method gets called over and over again, this will keep allocating new basemaps without cleaning any up. we should probably do some checks to see if it is null and if not, delete it first.

```suggestion
  if (basemap)
    delete basemap;
    
  basemap = new Basemap(BasemapStyle::OsmLightGray, basemapStyleParameters, this);
```

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:67]: probably unnecessary to include checking in this logging for the map load status. I suggest removing it

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (ldanzinger) [reply ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:59]: Also, I would move the Map initialization to the constructor

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp:40]: I'd move the member initialization to the initializer list instead of in the body

```suggestion
ConfigureBasemapStyleLanguage::ConfigureBasemapStyleLanguage(QObject* parent /* = nullptr */) :
  QObject(parent),
  m_map(new Map(SpatialReference::webMercator()),
  m_basemapStyleParameters(new BasemapStyleParameters(this))
{
}
```

**Review by ldanzinger — CHANGES_REQUESTED** [review]
Looking good, I've got a couple more suggestions for cleanup

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp
Comment (har13205) [reply ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.cpp]: No, the map initially launches with `local` language strategy

**Review by ldanzinger — APPROVED** [review]
LGTM. Looks like the design is approved and about to be merged. I think we might want to wait until it is officially merged before we merge this. You can watch the #native-samples-announcements channel for this

File: ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.qml
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/ConfigureBasemapStyleLanguage/ConfigureBasemapStyleLanguage.qml:124]: Per our offline discussion, I think this should be more intentional. It seems this executes when the code first loads, but it's unclear from looking at the c++ code how any basemap was loaded. I think the code should be more intentional about how the initial basemap is being set.

When I reviewed the latest code, I assumed no basemap at all was set, because that's what is happening on the C++ side via the constructor.

**Review by JamesMBallard — APPROVED** [review]

**Pull request merged**
